### PR TITLE
Remove ExecutionContext.Implicits.global from main library code

### PR DESCRIPTION
**Pull request by solicode**
After upgrading to 6.0.4, I noticed lots of strange issues in one of my Play apps with futures (timeouts, thread starvation, etc). It looks like the issue is coming from the use of `scala.concurrent.ExecutionContext.Implicits.global` in `HttpClient` (Play/Akka uses its own ExecutionContext, and it's problematic to add Implicits.global into the mix).

This PR is for addressing that by making the ExecutionContext an implicit parameter so that the caller has the option to specify it.

Also, I noticed `HttpClient.execute` is using a Promise which seems unnecessary since we're already working with a Future. I updated that too to get rid of the unnecessary overhead.

Unfortunately I had to touch pretty much all the tests too. I just used `scala.concurrent.ExecutionContext.Implicits.global` there as it doesn't matter for the tests.

### CODE REVIEW

File: elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/HttpClient.scala [hunk 1]
```diff
@@ -59,24 +56,14 @@ trait HttpClient extends Logging {
...
-            p.trySuccess(Left(failure))
-        } catch {
-          case t: Throwable => p.tryFailure(t)
-        }
-      case Failure(t) => p.tryFailure(t)
-    }
-    p.future
+  def execute[T, U](request: T)(implicit exec: HttpExecutable[T, U], executionContext: ExecutionContext): Future[Either[RequestFailure, RequestSuccess[U]]] = {
+    exec.execute(client, request)
+      .map(r => exec.responseHandler.handle(r) match {
```
Comment (sksamuel) [inline elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/HttpClient.scala]: I don't think this handles the case where the response handler throws an exception, which I'd like to see materialized as a Future.failure like before.

**Comment by sksamuel** [comment]
In general this change is probably required, but I'm relutant to bring it into 6.0.x and break everyone's code as they'll now need the implicit in scope. What about making it the default for the implicit and in 6.1 we drop the default?

File: elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/HttpClient.scala (hunk 1 above)
Comment (solicode) [reply elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/HttpClient.scala]: It wouldn't be safe in `onComplete`, but I don't think exceptions in `map` escape the context of Future.

```scala
val f = Future.successful("not a number").map(_.toInt)
f.failed.map(_ => println("failed future"))
```

results in a proper Future.failed.

https://docs.scala-lang.org/overviews/core/futures.html

> If the mapping function throws an exception the future is completed with that exception.

I think the 2 pieces of code are functionally equivalent, unless I'm forgetting to take something into account.

**Comment by solicode** [comment]
Sure, the default parameter idea sounds reasonable for 6.0.x. I updated the code.

File: elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/HttpClient.scala (hunk 1 above)
Comment (sksamuel) [reply elastic4s-http/src/main/scala/com/sksamuel/elastic4s/http/HttpClient.scala]: Ok

**Comment by sksamuel** [comment]
👍

**Pull request merged**